src/main/java/io/fabric8/jenkins/openshiftsync/BuildDecisionHandler.java

```diff
@@ -107,26 +111,33 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
         logger.info("Triggering OpenShift Build for WorkflowJob " + wj.getFullName() + " due to " + causeDescription(actions));
         BuildConfig foundBuildConfig = OpenShiftUtils.findBCbyNameOrLabel(getOpenShiftClient(), namespace, buildConfigname);
         if (foundBuildConfig != null) {
+          int buildInJenkins = wj.getLastBuild()!=null ? wj.getLastBuild().number : 0;
           buildConfigname = foundBuildConfig.getMetadata().getName();
-          triggerBuildRequest(buildConfigname, namespace, jobURL);
+          if (isPRBuild && cancelPRBuildOnUpdate && (buildInJenkins > 0 || (buildInJenkins == 0 && isJobInQueue(wj)))) {
```

Comment:
instead of nesting, you may be able to make it flatter (easier to understand) if you exit early; e.g.

```java
if (foo) {
  something()
  if (bar) {
    somethingElse();
      if (bar) {
        someMoreThings()
      }
  }
}
```

can be better written as 
```java

if (!foo) {
  return 
}
something()

if (!bar) {
return
}
somethingElse()

if (!baz) {
  return
}

someMoreThings();

```